src/app/clusters/account-login-server/account-login-server.cpp

```diff
@@ -136,41 +136,50 @@ bool emberAfAccountLoginClusterLoginCallback(app::CommandHandler * command, cons
 {
     CHIP_ERROR err               = CHIP_NO_ERROR;
     EndpointId endpoint          = commandPath.mEndpointId;
+    EmberAfStatus status         = EMBER_ZCL_STATUS_SUCCESS;
     auto & tempAccountIdentifier = commandData.tempAccountIdentifier;
     auto & setupPin              = commandData.setupPIN;
 
     Delegate * delegate = GetDelegate(endpoint);
     VerifyOrExit(isDelegateNull(delegate, endpoint) != true, err = CHIP_ERROR_INCORRECT_STATE);
 
+    if (!delegate->HandleLogin(tempAccountIdentifier, setupPin))
+    {
+        status = EMBER_ZCL_STATUS_NOT_AUTHORIZED;
+    }
+
 exit:
     if (err != CHIP_NO_ERROR)
     {
         ChipLogError(Zcl, "emberAfAccountLoginClusterLoginCallback error: %s", err.AsString());
         emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_FAILURE);
```

Comment:
if err is not CHIP_NO_ERROR this seems to send emberAfSendImmediateDefault report twice. Is this intentional? I would assume no.

Reply:
Same comment for methods below.